### PR TITLE
filter: aws ec2 tags

### PR DESCRIPTION
**Pull request by mwarzynski**
https://github.com/fluent/fluent-bit/pull/6065

### CODE REVIEW

**Review by PettitWesley — APPROVED** [review]
@mwarzynski This satisfies the requirement. It might be nice to add an section showing what the log record will look like once tags are added.

**Comment by PettitWesley** [comment]
@mwarzynski please remind me to merge this as soon as there is a new release. Your change should go out in the next.

**Comment by mwarzynski** [comment]
I added a more extensive documentation for the EC2 Tags feature. It covers all configuration options related to this feature (`tags_include` and `tags_exclude` too). For this reason, I think it would be great to get another review.

File: pipeline/filters/aws-metadata.md
Comment (PettitWesley) [inline pipeline/filters/aws-metadata.md:82]: I think this is too repetitive, let's not have the config options here agian, let's just note the users can read them in the main config section

File: pipeline/filters/aws-metadata.md
Comment (PettitWesley) [inline pipeline/filters/aws-metadata.md:96]: since you have a tags include and tags exclude example below, I think this is repetitive and not needed, user can just read those sections, which have examples of the output which is great

File: pipeline/filters/aws-metadata.md
Comment (PettitWesley) [inline pipeline/filters/aws-metadata.md:114]: 😎

File: pipeline/filters/aws-metadata.md
Comment (PettitWesley) [inline pipeline/filters/aws-metadata.md:74]: I think this is too repetitive, let's not have the config options here again, let's just note the users can read them in the main config section

**Review by PettitWesley — CHANGES_REQUESTED** [review]

File: pipeline/filters/aws-metadata.md
Comment (mwarzynski) [reply pipeline/filters/aws-metadata.md]: Sure, I deleted the 'Configuration' from the EC2 Tags section. You're right that our table with all config options should be sufficient.

File: pipeline/filters/aws-metadata.md
Comment (mwarzynski) [reply pipeline/filters/aws-metadata.md]: Yeah, you're right. It's repetitive. I removed this example.

**Comment by mwarzynski** [comment]
> please remind me to merge this as soon as there is a new release. Your change should go out in the next.

@PettitWesley These changes were released in the v2.1.0. https://github.com/fluent/fluent-bit/releases/tag/v2.1.0

I think we are good to go and merge this PR too! (Unless you can find more improvements in the proposed docs.)

File: pipeline/filters/aws-metadata.md
Comment (PettitWesley) [inline pipeline/filters/aws-metadata.md:72]: I think this line is unnecessary

Comment (PettitWesley) [reply pipeline/filters/aws-metadata.md:72]: Especially thanks to the awesome config example with log output right below this line

Comment (mwarzynski) [reply pipeline/filters/aws-metadata.md]: > I think this line is unnecessary

I removed this line. It should be fine now.

File: pipeline/filters/aws-metadata.md
Comment (PettitWesley) [inline pipeline/filters/aws-metadata.md:70]: > Fluent Bit will operate as if there are no tags associated with the EC2 instance.

@mwarzynski  Can you be more clear that Fluent Bit will not exit/fail but will run without adding tags.

**Review by PettitWesley — APPROVED** [review]
One minor comment but approving

File: pipeline/filters/aws-metadata.md
Comment (mwarzynski) [reply pipeline/filters/aws-metadata.md]: > To use the `tags_enabled true` functionality in Fluent Bit, the [instance-metadata-tags](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/modify-instance-metadata-options.html) option must be enabled on the EC2 instance where Fluent Bit is running. Without this option enabled, Fluent Bit will not be able to retrieve the tags associated with the EC2 instance. However, this does not mean that Fluent Bit will fail or stop working altogether. Instead, if [instance-metadata-tags](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/modify-instance-metadata-options.html) option is not enabled, Fluent Bit will continue to operate normally and capture other values, such as the EC2 instance ID or availability zone, based on its configuration.

@PettitWesley Is it okay now?

**Review by PettitWesley — APPROVED** [review]
This looks really good now! Thank you!

**Pull request merged**
